backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ir/NewIrUtils.kt

```diff
@@ -22,22 +22,32 @@ import org.jetbrains.kotlin.ir.types.isMarkedNullable
 import org.jetbrains.kotlin.ir.util.*
 import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.resolve.lazy.NoTopLevelDescriptorProvider.getPackageFragment
 import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-val IrField.fqNameForIrSerialization: FqName get() = this.parent.fqNameForIrSerialization.child(this.name)
+fun IrClass.isUnit() = this.hasEqualFqName(StandardNames.FqNames.unit.toSafe())
 
-fun IrClass.isUnit() = this.fqNameForIrSerialization == StandardNames.FqNames.unit.toSafe()
-
-fun IrClass.isKotlinArray() = this.fqNameForIrSerialization == StandardNames.FqNames.array.toSafe()
+fun IrClass.isKotlinArray() = this.hasEqualFqName(StandardNames.FqNames.array.toSafe())
 
 val IrClass.superClasses get() = this.superTypes.map { it.classifierOrFail as IrClassSymbol }
 fun IrClass.getSuperClassNotAny() = this.superClasses.map { it.owner }.atMostOne { !it.isInterface && !it.isAny() }
 
-fun IrClass.isAny() = this.fqNameForIrSerialization == StandardNames.FqNames.any.toSafe()
-fun IrClass.isNothing() = this.fqNameForIrSerialization == StandardNames.FqNames.nothing.toSafe()
+fun IrClass.isAny() = this.hasEqualFqName(StandardNames.FqNames.any.toSafe())
+
+fun IrDeclarationWithName.hasEqualFqName(fqName: FqName): Boolean {
+    val comparisonResult = this.name == fqName.shortName()
+    if (!comparisonResult || (parent is IrPackageFragment && (parent as IrPackageFragment).fqName == fqName.parent()))
```

Comment:
Please split this to two ifs. Current version is somewhat hard to read.
Also why not return false immediatly if parent is package fragment but has unmatching fq name?